medcat/utils/ner/deid.py

```diff
@@ -62,18 +62,73 @@ def train(self, json_path: Union[str, list, None],
               *args, **kwargs) -> Tuple[Any, Any, Any]:
         return super().train(json_path, *args, train_nr=0, **kwargs)  # type: ignore
 
-    def deid_text(self, text: str, redact: bool = False) -> str:
+    def get_chunks(self, text:str, de_id_pipe,maximum_tkns) -> List[List[Any]]:
+
+        blocks = [[]]
+        if 'Roberta' in de_id_pipe.ner_pipe.tokenizer.__class__.__name__:
+
+            whitespace_pattern = re.compile(r'\s')
+            tok_output = de_id_pipe.ner_pipe.tokenizer(text, return_offsets_mapping=True)
+            for token, (start, end) in zip(tok_output['input_ids'], tok_output['offset_mapping']):
+                if token in [0, 2]:
+                    continue
+                if len(blocks[-1]) == maximum_tkns:
+                    to_append_block = []
+                    idx_chunk = -1
+                    for i in range(len(blocks[-1]) - 1, len(blocks[-1]) - 21, -1):
```

Comment:
I'm still not sure what this magic number of `21` is. From the comment below it seems like it's got something to do with sub-word division, but I don't know what.